### PR TITLE
Refactoring and improvement of automotive documentation

### PR DESCRIPTION
**Pull request by polybassa**
I started to refactor automotive related documentation. This is just the start. I will do more refactoring in 2020.

### CODE REVIEW

File: doc/scapy/layers/automotive.rst [hunk 1]
```diff
@@ -66,88 +75,136 @@ Send a message over a Vector CAN-Interface::
...
+^^^^^^^^^^^^^^^
+
+In Scapy, two kind of CANSockets are implemented. One implementation is called **Native CANSocket**,
+the other implementation is called **Python-can CANSocket**.
+
+Since Python 3 supports ``PF_CAN`` sockets, **Native CANSockets** can be used on a
+Linux based system with Python 3 or higher. These sockets have a performance advantage
+because ``select`` is callable on them. This has a big effect in MITM scenarios.
+
+For compatibility reasons, **Python-can CANSockets** were add to Scapy.
```
Comment (guedou) [inline doc/scapy/layers/automotive.rst]: s/add/added/

Comment (polybassa) [reply doc/scapy/layers/automotive.rst]: Done. Thanks

File: doc/scapy/layers/automotive.rst [hunk 2]
```diff
@@ -66,88 +75,136 @@ Send a message over a Vector CAN-Interface::
...
+related contributions.
 
+Virtual CAN sockets require a special Linux kernel module. The following shell command loads the required module::
 
-CAN Layer
----------
+    sudo modprobe vcan
+
+In order to use a virtual CAN interface some additional commands for setup are required.
+This snippet chooses the name ``vcan0`` for the virtual CAN interface. Any name can be choosen here::
```
Comment (guedou) [inline doc/scapy/layers/automotive.rst]: codespell complains about `choosen`. It should be `chosen`.

**Comment by guedou** [comment]
PR #2388 adds the requested fix.

**Comment by polybassa** [comment]
Thanks a lot. Sorry for the delay.

**Pull request closed without merging**